rfcs/0134-nix-store-layer.md

```diff
@@ -0,0 +1,271 @@
+[examples-and-interactions]: #examples-and-interactions
+
+## Maximal Nix unlike today
+
+I cannot emphasize this enough, but the interface of maximal store+exprs+flakes Nix remains *exactly* like today.
+In particular, lower level commands can be used with higher level "installables" (arguments), so e.g.
+```
+nix show-derivation flake#bar
+```
+will still work.
+
+## Good tasks anyways!
+
+Lots of the plan above I think is good work we should be doing anyways, regardless of whether we expose a store-only Nix.
+If you believe this, then the "cost" of this RFC is a lot less.
+
+### Tests
+
+Splitting the test suite per natural layer of the implementation is good work because it combines the specificity of unit tests with the real-world-ness of integration tests.
+"entire kitchen sink" tests make it harder to narrow down root causes of failures.
+
+### Manual
+
+The documentation team is already working to clean up the manual, and this effort already involves emphasizing layering.
+So whether we formally make a store-only Nix or not, I suspect the overhauled manual will natural have easy boundaries from which to "extract" the store-only manual.
+
+## Security
+
+The daemon is a privileged process.
+Even if with upcoming changes it shouldn't need root, it does tasks like administrating OS sandboxes correctly which still are security-critical.
+Having less code in the story-only Nix daemon, even if we think the removed code was "dead anyways" is always good.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Creating new teams, trying to build ties with other communities sounds scary.
+
+# Alternatives
+[alternatives]: #alternatives
+
+Do programming parts without the documentation and website parts.
+But that feels to me like turning an ongoing shift in focus to a one-off change that is likely to bit-rot.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+What should the store-only Nix be called?
+
+# Future work
+[future]: #future-work
+
+## Nix Store Team
+
+Now that we have this division in the implementation, we also have the opportunity to leverage it for governance purposes.
+An official, NixOS-foundation-authorized team could be set up to manage store layer design decisions (below the threshold of needing an RFC) that don't effect that the rest of Nix.
+\[Some sort of decision that affect all layers is out of scope, must be deliberated with stakeholders from other layers too, probably should be RFC due to such large scope.\]
+
+To be clear, this is *not* to say we should abandon the idea of Nix as a whole.
+There can still be governance of Nix as a whole; this team, and similar hypothetical, say, Flakes, Nix language, or User Experience teams would ultimately need to report to.
+The goal is not to overreact, but strike a balance between:
+
+1. Making sure Nix as a whole continues to make sense
+2. Make sure layers make sense in isolation not just in the context of the way they are currently used.
+
+## Standardization across projects
```

Comment:
Have you reached out to Guix yet? Or if not, when do you think it would be valuable to get their input?

Reply:
I have! But I think for this RFC the ball is purely in our court. This RFC is not about making technical decisions that would impact Guix, but about ratifying the idea of a store layer that is usable in arbitrary ways including the Nix language and Scheme via Guix.

I'll continue talking to them, and I encourage anyone else that wants to reach and say hi too — let's build more bridges between our communities — but if we don't do something like this I don't see why they should take us very seriously about store layer interop.